app/lib/screens/login_screen.dart

```diff
@@ -231,7 +234,25 @@ class LoginScreenState extends State<LoginScreen> {
                         icon: const Icon(Icons.qr_code_scanner),
                         label: Text(
                             AppLocalizations.of(context)!.qrCodeScanButton),
-                        onPressed: () {},
+                        onPressed: () async {
+
+                          final scannedQR = await Navigator.push(
```

Comment:
You could also assert the type here: `await Navigator.push<String?>` but I'm not sure if that's safe or could throw an error.